portal-impl/src/com/liferay/portal/service/impl/WorkflowDefinitionLinkLocalServiceImpl.java

```diff
@@ -382,6 +408,41 @@ public WorkflowDefinitionLink updateWorkflowDefinitionLink(
 		return workflowDefinitionLinkPersistence.update(workflowDefinitionLink);
 	}
 
+	@Override
+	public WorkflowDefinitionLink updateWorkflowDefinitionLink(
+			Long groupId, String className,
+			WorkflowDefinitionLink oldWorkflowDefinitionLink)
+		throws PortalException {
+
+		User user = _userPersistence.findByPrimaryKey(
+			oldWorkflowDefinitionLink.getUserId());
+
+		WorkflowDefinitionLink workflowDefinitionLink =
+			workflowDefinitionLinkPersistence.findByERC_G(
+				oldWorkflowDefinitionLink.getExternalReferenceCode(), groupId);
+
+		if (workflowDefinitionLink != null) {
+			workflowDefinitionLinkPersistence.remove(workflowDefinitionLink);
+		}
+
+		WorkflowDefinitionLink newWorkflowDefinitionLink =
+			addWorkflowDefinitionLink(
+				oldWorkflowDefinitionLink.getUserId(),
+				oldWorkflowDefinitionLink.getCompanyId(),
+				StagingUtil.getLiveGroupId(groupId), className,
+				oldWorkflowDefinitionLink.getClassPK(),
+				oldWorkflowDefinitionLink.getTypePK(),
+				oldWorkflowDefinitionLink.getWorkflowDefinitionName(),
+				oldWorkflowDefinitionLink.getWorkflowDefinitionVersion());
+
+		newWorkflowDefinitionLink.setUserName(user.getFullName());
```

Comment:
Why are you doing this extra logic here and calling the `workflowDefinitionLinkPersistence.update` again?

Inside the `addWorkflowDefinitionLink` method we already have this logic, including the update being called as you can see [here](https://github.com/liferay/liferay-portal/blob/9286cbb909d0eb4cf3480a7b52b4d9a98c3b438c/portal-impl/src/com/liferay/portal/service/impl/WorkflowDefinitionLinkLocalServiceImpl.java#L40-L67).

```
User user = _userPersistence.findByPrimaryKey(userId);
workflowDefinitionLink.setUserName(user.getFullName());
workflowDefinitionLink.setClassNameId(
	_classNameLocalService.getClassNameId(className));
```